diff --git a/Linux_for_Tegra/source/nvidia-oot/drivers/media/i2c/max96712.c b/Linux_for_Tegra/source/nvidia-oot/drivers/media/i2c/max96712.c
index 8e237eef0..27b8662aa 100644
--- a/Linux_for_Tegra/source/nvidia-oot/drivers/media/i2c/max96712.c
+++ b/Linux_for_Tegra/source/nvidia-oot/drivers/media/i2c/max96712.c
@@ -30,6 +30,10 @@ int max96712_read_reg_Dser(int slaveAddr, int channel, u16 addr, unsigned int *v

 struct mutex max96712_rw;

+#ifdef CONFIG_VIDEO_ECAM
+int total_connected_cam, ret_val, des_channel = 0;
+#endif
+
 int max96712_write_reg_Dser(int slaveAddr, int channel,
 				u16 addr, u8 val)
 {
@@ -153,6 +157,7 @@ static const struct file_operations max96712_debugfs_fops = {
 	.release = single_release,
 };

+#ifndef CONFIG_VIDEO_ECAM
 static int  max96712_power_on(struct max96712 *priv)
 {
 	struct i2c_client *i2c_client = priv->i2c_client;
@@ -170,6 +175,7 @@ static int  max96712_power_on(struct max96712 *priv)
 	}
 	return 0;
 }
+#endif

 static int max96712_debugfs_init(const char *dir_name,
 				struct dentry **d_entry,
@@ -197,6 +203,10 @@ static int max96712_debugfs_init(const char *dir_name,
 		return -EINVAL;
 	global_priv[index] = priv;

+#ifdef CONFIG_VIDEO_ECAM
+	des_channel = index;
+#endif
+
 	dev_dbg(&i2c_client->dev, "%s: index %d\n", __func__, index);

 	dp = debugfs_create_dir(dev_name, NULL);
@@ -251,23 +261,51 @@ static int max96712_probe(struct i2c_client *client,
 	}

 	mutex_init(&max96712_rw);
-
+#ifndef CONFIG_VIDEO_ECAM
 	err = max96712_power_on(priv);
 	if (err) {
 		dev_err(&client->dev, "Failed to power on err =%d\n",err);
 		return err;
 	}
-
+#endif
 	err = max96712_debugfs_init(NULL, NULL, NULL, priv);
 	if (err)
 		return err;

+#ifdef CONFIG_VIDEO_ECAM
+	err = max96712_write_reg_Dser(0x52, des_channel, 0x0006, 0xFF);
+
+	err = max96712_read_reg_Dser(0x52, des_channel, 0x001A, &ret_val);
+	if(ret_val & 0x08)
+		total_connected_cam++;
+	ret_val = 0;
+
+	err = max96712_read_reg_Dser(0x52, des_channel, 0x000A, &ret_val);
+	if(ret_val != 0)
+		total_connected_cam++;
+	ret_val = 0;
+
+	err = max96712_read_reg_Dser(0x52, des_channel, 0x000B, &ret_val);
+	if(ret_val & 0x08)
+		total_connected_cam++;
+	ret_val = 0;
+
+	err = max96712_read_reg_Dser(0x52, des_channel, 0x000C, &ret_val);
+	if(ret_val & 0x08)
+		total_connected_cam++;
+	ret_val = 0;
+#endif
+
 	/*set daymode by fault*/
 	dev_info(&client->dev, "%s:  success\n", __func__);

 	return err;
 }

+#ifdef CONFIG_VIDEO_ECAM
+EXPORT_SYMBOL(total_connected_cam);
+#endif
+
 #if defined(NV_I2C_DRIVER_STRUCT_REMOVE_RETURN_TYPE_INT) /* Linux 6.1 */
 static int max96712_remove(struct i2c_client *client)
 #else
@@ -282,6 +320,9 @@ static void max96712_remove(struct i2c_client *client)
 #if defined(NV_I2C_DRIVER_STRUCT_REMOVE_RETURN_TYPE_INT) /* Linux 6.1 */
 	return 0;
 #endif
+#ifdef CONFIG_VIDEO_ECAM
+	return 0;
+#endif
 }

 static const struct i2c_device_id max96712_id[] = {
